src/main/java/com/xceptance/xlt/mastercontroller/Main.java

```diff
@@ -490,6 +519,44 @@ protected boolean validateCommandLine(final CommandLine commandLine)
             }
         }
 
+        // check --only-download option
+        if (commandLine.hasOption(OPTION_DOWNLOAD))
+        {
+            // --only-download and -noDownload are mutually exclusive
+            if (commandLine.hasOption(OPTION_NO_DOWNLOAD))
+            {
+                final String message = String.format("Option '--%s' cannot be used together with option '-%s'.", OPTION_DOWNLOAD,
+                                                     OPTION_NO_DOWNLOAD);
+                System.out.println(message);
+                log.error(message);
+
+                invalid = true;
+            }
+            else if (!(commandMode || autoMode || sequentialMode || fafMode))
+            {
+                final String message = String.format("Option '--%s' can only be used in non-interactive mode.", OPTION_DOWNLOAD);
+                System.out.println(message);
+                log.error(message);
+
+                invalid = true;
+            }
+            else
+            {
+                // validate option values
```

Comment:
Maybe we need to check first if the option value has a value at all. 
Consider passing the option in the following variants:

```
--only-download=
--only-download ""
```

In these cases, the download currently fails with an exception:

```
Failed to run master controller: Downloading test results failed.
```

Reply:
Improved validation of download option.